PowerArubaCX/Public/VM.ps1

```diff
@@ -0,0 +1,357 @@
+
+        PS>Deploy-ArubaCXVm @cxBuildParams
+
+        Deploy Aruba CX VM by pass array with settings.
+    #>
+
+    Param(
+        [Parameter (Mandatory = $true, Position = 1)]
+        [string]$ovf_path,
+        [Parameter (Mandatory = $true, Position = 2)]
+        [string]$vm_host,
+        [Parameter (Mandatory = $true)]
+        [string]$datastore,
+        [Parameter (Mandatory = $true)]
+        [string]$cluster,
+        [Parameter (Mandatory = $false)]
+        [string]$inventory,
+        [Parameter (Mandatory = $true)]
+        [string]$name_vm,
+        [Parameter (Mandatory = $false)]
+        [ValidateRange(1, 32)]
+        [int]$memoryGB,
+        [Parameter (Mandatory = $false)]
+        [ValidateRange(1, 32)]
+        [int]$cpu,
+        [Parameter(Mandatory = $false)]
+        [switch]$StartVM = $false,
+        [Parameter (Mandatory = $true)]
+        [string]$vmnetwork
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        #Check if VMWare PowerCLI is available (not use #Require because not mandatory module)
+        if ($null -eq (Get-InstalledModule -name VMware.VimAutomation.Common -ErrorAction SilentlyContinue)) {
+            Throw "You need to install VMware.PowerCLI (Install-Module VMware.PowerCLI)"
+        }
+        #Write-Warning "You need to have a vSwitch configured on your vSphere environment even if you use a DVS"
+        #default vapp_config
+        $vapp_config = @{
+            "source" = $ovf_path
+            "name"   = $name_vm
+        }
+
+        if ($DefaultVIServers.Count -eq 0) {
+            throw "Need to be connect to vCenter (use Connect-VIServer)"
+        }
+        if (Get-VM $name_vm -ErrorAction "silentlyContinue") {
+            Throw "VM $name_vm already exist, change name or remove VM"
+        }
+
+        if (-not (Get-Cluster -Name $cluster -ErrorAction "silentlycontinue")) {
+            Throw "Cluster not found : $cluster"
+        }
+        else {
+            $vapp_config.add("Location", $cluster)
+        }
+
+        if (-not (Get-VMHost -Name $vm_host -ErrorAction "silentlycontinue")) {
+            Throw "Vm_Host not found : $vm_host"
+        }
+        else {
+            $vapp_config.add("vmhost", $vm_host)
+        }
+
+        if (-not (Get-Datastore -Name $datastore -ErrorAction "silentlycontinue")) {
+            Throw "Datastore not found : $datastore"
+        }
+        else {
+            $vapp_config.add("datastore", $datastore)
+        }
+
+        if ( $PsBoundParameters.ContainsKey('inventory') ) {
+            if (-not (Get-Inventory -Name $inventory -ErrorAction "silentlycontinue")) {
+                Throw "Inventory not found : $inventory"
+            }
+            else {
+                $vapp_config.add("inventory", $inventory)
+            }
+        }
+
+        $ovfConfig = Get-OvfConfiguration -Ovf $ovf_path
+
+        #Check if vSwitch is available ?
+        $ovfConfig.NetworkMapping.Null.Value = $vmnetwork
```

Comment:
Is it necessary ? We can bound directly to DVS if I remember correctly, you just can"t deploy OVF if there's no vswitch on the VMWare infrastructure

Reply:
Yes it is the case but the idea it is to check if the vSwitch is available or not...